layers.go

```diff
@@ -12,54 +12,67 @@ import (
 
 type bpLayersDir struct {
 	path   string
-	layers []bpLayer
+	layers map[string]bpLayer
 	name   string
 }
 
 func readBuildpackLayersDir(layersDir, buildpackID string) (bpLayersDir, error) {
 	path := filepath.Join(layersDir, buildpackID)
-	tomls, err := filepath.Glob(filepath.Join(path, "*.toml"))
-	if err != nil {
-		return bpLayersDir{}, err
-	}
-	layers := make([]bpLayer, 0, len(tomls))
-	for _, toml := range tomls {
-		name := strings.TrimRight(filepath.Base(toml), ".toml")
-		layers = append(layers, bpLayer{
-			layer{
-				path:       filepath.Join(layersDir, buildpackID, name),
-				identifier: fmt.Sprintf("%s/%s", buildpackID, name),
-			},
-		})
-	}
-	return bpLayersDir{
+	layers := map[string]bpLayer{}
```

Comment:
If `layers` is extracted, use it later instead of `bpDir.layers`. Otherwise, I suggest inlining it.